pkg/agent/cniserver/server.go

```diff
@@ -492,6 +507,18 @@ func (s *CNIServer) CmdDel(_ context.Context, request *cnipb.CniCmdRequest) (
 		return s.configInterfaceFailureResponse(err), nil
 	}
 	klog.Infof("CmdDel for container %v succeeded", cniConfig.ContainerId)
+
+	if s.secondaryNetworkEnabled {
+		podName := string(cniConfig.K8S_POD_NAME)
+		podNamespace := string(cniConfig.K8S_POD_NAMESPACE)
+
+		containerInfo := s.podConfigurator.podInfoStore.GetCNIConfigInfoByContaninerID(podName, podNamespace, cniConfig.ContainerId)
+		if containerInfo != nil {
+			// Update PodCNIDeleted = true.
+			// This is required to release network specific IP address at whereabouts DB.
```

Comment:
Guess we can update the comment to say it is for Pod controller to know the Pod has been deleted from CNI.